src/gluonts/dataset/dataframe.py

```diff
@@ -0,0 +1,300 @@
+# permissions and limitations under the License.
+
+from copy import deepcopy
+from dataclasses import dataclass, field
+from datetime import datetime
+from typing import Iterator, List, Union, Dict
+
+import pandas as pd
+
+from gluonts.dataset.common import Dataset, DataEntry, ProcessDataEntry
+from gluonts.dataset.field_names import FieldName
+
+
+@dataclass
+class DataFramesDataset(Dataset):
+    """
+    A pandas.DataFrame-based dataset type.
+
+    This class is constructed with a collection of pandas.DataFrame-objects
+    where each ``DataFrame`` is representing one time series.
+    A ``target`` and a ``timestamp`` columns are essential. Furthermore,
+    static/dynamic real/categorical features can be specified.
+
+    Parameters
+    ----------
+    dataframes: Union[
+        List[pandas.DataFrame], Dict[str, pandas.DataFrame], pandas.DataFrame
+    ]
+        List or Dict of ``DataFrame``s or a single ``DataFrame`` containing
+        at least ``timestamp`` and ``target`` columns. If a Dict is provided,
+        the key will be the associated ``item_id``.
+    target: str or List[str]
+        Name of the column that contains the ``target`` time series.
+        For multivariate targets, a list of column names should be provided.
+    timestamp: str
+        Name of the column that contains the timestamp information.
+    freq: str
+        Frequency of observations in the time series. Must be a valid pandas
+        frequency.
+    feat_dynamic_real: List[str]
+        List of column names that contain dynamic real features.
+    feat_dynamic_cat: List[str]
+        List of column names that contain dynamic categorical features.
+    feat_static_real: List[str]
+        List of column names that contain static real features.
+    feat_static_cat: List[str]
+        List of column names that contain static categorical features.
+    past_feat_dynamic_real: List[str]
+        List of column names that contain dynamic real features only for the
+        history.
+    ignore_last_n_targets: int
+        For target and past dynamic features last ``ignore_last_n_targets``
+        elements are removed when iterating over the data set. This becomes
+        important when the predictor is called.
+    """
+
+    dataframes: Union[
+        List[pd.DataFrame], Dict[str, pd.DataFrame], pd.DataFrame
+    ]
+    target: Union[str, List[str]]
+    timestamp: str
+    freq: str
+    feat_dynamic_real: List[str] = field(default_factory=list)
+    feat_dynamic_cat: List[str] = field(default_factory=list)
+    feat_static_real: List[str] = field(default_factory=list)
+    feat_static_cat: List[str] = field(default_factory=list)
+    past_feat_dynamic_real: List[str] = field(default_factory=list)
+    ignore_last_n_targets: int = 0
+
+    def __post_init__(self) -> None:
+        if isinstance(self.target, list) and len(self.target) == 1:
+            self.target = self.target[0]
```

Comment:
This will not allow for Multivariate-1-dimensional target. I.e. no support for matrix of size (1, T). It will always be represented as a vector of length T. Is that desired?

Reply:
I think this is ok